src/Git.hs

```diff
@@ -349,46 +345,46 @@ runGit userConfig repoDir operation =
         Right newSha ->
           pure $ Just $ Sha $ Text.stripEnd newSha
 
-  in case operation of
-    FetchBranch branch withTags cont -> do
+  in interpret $ \_ -> \case
+    FetchBranch branch withTags -> do
       result <- callGitInRepo $ case withTags of
         WithTags -> ["fetch", "--tags", "origin", refSpec branch]
         NoTags   -> ["fetch", "origin", refSpec branch]
       case result of
         Left (_, message) -> logWarnN $ "warning: git fetch failed. Reason: " <> message
         Right _ -> return ()
-      pure cont
 
-    ForcePush sha branch cont -> do
+
+    ForcePush sha branch -> do
       -- TODO: Make Sha and Branch constructors sanitize data, otherwise this
       -- could run unintended Git commands.
       -- Note: the remote branch is prefixed with 'refs/heads/' to specify the
       -- branch unambiguously. This will make Git create the branch if it does
       -- not exist.
       gitResult <- callGitInRepo ["push", "--force", "origin", refSpec (sha, branch)]
       case gitResult of
-        Right _ -> pure $ cont PushOk
+        Right _ -> pure PushOk
         Left (_, message) -> do
           logWarnN $ "error: git push --force failed. Reason: " <> message
-          pure $ cont $ PushRejected message
+          pure $ PushRejected message
 
-    Push sha branch cont -> do
+    Push sha branch -> do
       result <- callGitInRepo ["push", "origin", refSpec (sha, branch)]
       case result of
         Left  (_, message) -> do
           logInfoN $ "warning: git push failed. Reason: " <> message
-          pure . cont $ PushRejected message
-        Right _ -> pure $ cont PushOk
+          pure $ PushRejected message
+        Right _ -> pure PushOk
 
-    DeleteRemoteBranch branch cont -> do
+    DeleteRemoteBranch branch -> do
       gitResult <- callGitInRepo ["push", "origin", "-d", refSpec branch]
       case gitResult of
-        Right _ -> pure $ cont PushOk
+        Right _ -> pure PushOk
         Left (_, message) -> do
           logWarnN $ "error: git push -d failed. Reason: " <> message
-          pure $ cont $ PushRejected message
+          pure $ PushRejected message
 
-    Rebase sha remoteBranch cont -> do
+    Rebase sha remoteBranch -> do
```

Comment:
This shows the removal of those `cont` arguments and calls. It may be nitpicking, but I like not having to deal with them.

Reply:
I don't think it's nitpicking. When you express something with code, you don't want boilerplate, so unwarranted CPS is not really your friend here :smiley: